pkg/logs/client/tcp/connection_manager.go

```diff
@@ -121,6 +123,17 @@ func (cm *ConnectionManager) NewConnection(ctx context.Context) (net.Conn, error
 	}
 }
 
+func (cm *ConnectionManager) handshakeWithTimeout(conn *tls.Conn, timeout time.Duration) error {
+	errChannel := make(chan error, 2)
+	time.AfterFunc(timeout, func() {
+		errChannel <- timeoutError{}
+	})
+	go func() {
+		errChannel <- conn.Handshake()
```

Comment:
as this call can hang out for ever, it means that this go function will leak ?

Reply:
Indeed the go routine will leak.

Reply:
I have added debug logs